prusti-contracts/src/lib.rs

```diff
@@ -84,8 +98,59 @@ mod private {
     /// A macro for defining a predicate using prusti expression syntax instead
     /// of just Rust expressions.
     pub use prusti_contracts_internal::predicate;
-}
 
+    /// A sequence type
+    #[non_exhaustive]
+    #[derive(PartialEq, Eq, Copy, Clone)]
+    pub struct Seq<T: Copy> {
+        _phantom: core::marker::PhantomData<T>,
+    }
+
+    impl<T: Copy> Seq<T> {
+        #[pure]
+        pub fn empty() -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn single(_: T) -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn concat(_: Self, _: Self) -> Self {
+            panic!()
+        }
+        #[pure]
+        pub fn lookup(self, index: usize) -> T {
+            panic!()
+        }
+    }
+
+    /// A map type
+    #[non_exhaustive]
+    #[derive(PartialEq, Eq, Copy, Clone)]
+    pub struct Map<K, V> {
+        _key_phantom: core::marker::PhantomData<K>,
+        _val_phantom: core::marker::PhantomData<V>,
+    }
+
+    impl<K, V> Map<K, V> {
+        pub fn empty() -> Self {
+            panic!()
+        }
+        pub fn insert(self, _key: K, _val: V) -> Self {
+            panic!()
+        }
+        pub fn delete(self, _key: K) -> Self {
+            panic!()
+        }
+    }
+
+    pub macro ghost {
+        ($($t:tt)*) => {
+            $($t)*
```

Comment:
Does this only take the first token or does it somehow implicitly use `tt` as well?

Reply:
`$t` captures all tokens that are given as arguments to the ghost macro, i.e. all token trees (tt)
I will rename it to avoid confusion.